applications/GeoMechanicsApplication/custom_elements/line_interface_element.h

```diff
@@ -0,0 +1,72 @@
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include "includes/element.h"
+#include "includes/ublas_interface.h"
+#include "integration_scheme.h"
+#include "stress_state_policy.h"
+
+namespace Kratos
+{
+
+class KRATOS_API(GEO_MECHANICS_APPLICATION) LineInterfaceElement : public Element
+{
+public:
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(LineInterfaceElement);
+
+    using Element::GeometryType;
+    using Element::PropertiesType;
+    using BaseType = Element;
+
+    LineInterfaceElement() = default;
+
+    LineInterfaceElement(IndexType                      NewId,
+                         const GeometryType::Pointer&   rGeometry,
+                         const PropertiesType::Pointer& rProperties);
+
+    void EquationIdVector(EquationIdVectorType& rResult, const ProcessInfo&) const override;
+    void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix, const ProcessInfo&) override;
+    void CalculateRightHandSide(VectorType& rRightHandSideVector, const ProcessInfo&) override;
+    void CalculateLocalSystem(MatrixType&        rLeftHandSideMatrix,
+                              VectorType&        rRightHandSideVector,
+                              const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateOnIntegrationPoints(const Variable<Vector>& rVariable,
+                                      std::vector<Vector>&    rOutput,
+                                      const ProcessInfo&      rCurrentProcessInfo) override;
+    void CalculateOnIntegrationPoints(const Variable<ConstitutiveLaw::Pointer>& rVariable,
+                                      std::vector<ConstitutiveLaw::Pointer>&    rOutput,
+                                      const ProcessInfo&) override;
+    using BaseType::CalculateOnIntegrationPoints;
+    void GetDofList(DofsVectorType& rElementalDofList, const ProcessInfo&) const override;
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override;
+
+    Element::Pointer Create(IndexType NewId, const NodesArrayType& rNodes, PropertiesType::Pointer pProperties) const override;
+    Element::Pointer Create(IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties) const override;
+
+private:
+    Element::DofsVectorType GetDofs() const;
+
+    std::vector<Matrix> CalculateLocalBMatricesAtIntegrationPoints() const;
+    std::vector<double> CalculateIntegrationCoefficients() const;
+    std::vector<Matrix> CalculateConstitutiveMatricesAtIntegrationPoints();
+    std::vector<Vector> CalculateRelativeDisplacementsAtIntegrationPoints(const std::vector<Matrix>& rLocalBMatrices) const;
+    std::vector<Vector> CalculateTractionsAtIntegrationPoints(const std::vector<Vector>& rRelativeDisplacements);
+
+    std::unique_ptr<IntegrationScheme>    mIntegrationScheme;
+    std::unique_ptr<StressStatePolicy>    mStressStatePolicy;
+    std::vector<ConstitutiveLaw::Pointer> mConstitutiveLaws;
```

Comment:
You are completely right, good catch! However, it is not too simple to save/load the StressStatePolicy and IntegrationScheme, since they are abstract classes and we want to instantiate the specific types upon load. We still need to work out the best way to do this (also for e.g. the normal UPwSmallStrainElement), so I'd propose to implement the save/load for the interface element when we found a way to do it correctly